sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_challenge_auth_policy.py

```diff
@@ -13,26 +13,37 @@
 requirements can change. For example, a vault may move to a new tenant. In such a case the policy will attempt the
 protocol again.
 """
+from typing import TYPE_CHECKING
 
-from azure.core.pipeline import PipelineRequest
 from azure.core.pipeline.policies import AsyncHTTPPolicy
-from azure.core.pipeline.transport import HttpResponse
 
-from . import ChallengeAuthPolicyBase, HttpChallenge, HttpChallengeCache
+from . import HttpChallengeCache
+from .challenge_auth_policy import _enforce_tls, _get_challenge_request, _update_challenge, ChallengeAuthPolicyBase
+
+if TYPE_CHECKING:
+    from typing import Any
+    from azure.core.credentials_async import AsyncTokenCredential
```

Comment:
Means you n azure-core>=1.2.0

Reply:
And actually, the min dep is indeed 1.2.0, since you won't be broken with your changes with 1.2.0 ;).